config/config-api/src/com/thoughtworks/go/config/PipelineTemplateConfig.java

```diff
@@ -67,12 +69,62 @@ public CaseInsensitiveString name() {
         return name;
     }
 
+    public void validateTree(ValidationContext validationContext, CruiseConfig preprocessedConfig) {
+        validate(validationContext);
+        validateParamsAndFetchTasks(preprocessedConfig);
+    }
+
+    private void validateParamsAndFetchTasks(CruiseConfig preprocessedConfig) {
```

Comment:
Can you try and profile this with a large enough config to see what kind of impact it adds.